app/assets/javascripts/discourse/app/components/sidebar/toggle-all-sections.gjs

```diff
@@ -0,0 +1,54 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+import { service } from "@ember/service";
+import DButton from "discourse/components/d-button";
+import { ADMIN_NAV_MAP } from "discourse/lib/sidebar/admin-nav-map";
+
+export default class ToggleAllSections extends Component {
+  @service sidebarState;
+  @service keyValueStore;
+  @tracked collapsedSections = this.sidebarState.collapsedSections;
+
+  get allSectionsExpanded() {
+    return ADMIN_NAV_MAP.every((adminNav) => {
+      const collapsedSidebarSectionKey = `sidebar-section-${this.sidebarState.currentPanel.key}-${adminNav.name}-collapsed`;
```

Comment:
No need to allocate the const, just do "includes(`sidebar-section-${this.sidebarState.currentPanel.key}-${adminNav.name}-collapsed`)" IMO